test/language/statements/async-generator/yield-star-async-next.js

```diff
@@ -0,0 +1,196 @@
+
+                  args[0]({
+                    get value() {
+                      log.push({ name: "get next value (2)" });
+                      return "next-value-2";
+                    },
+                    get done() {
+                      log.push({ name: "get next done (2)" });
+                      return true;
+                    }
+                  });
+                };
+              }
+            };
+          };
+        }
+      };
+    };
+  }
+};
+var asyncIterator = (async function*() {
+  log.push({ name: "before yield*" });
+  var v = yield* iter;
+  log.push({
+    name: "after yield*",
+    value: v
+  });
+  return "return-value";
+})();
+
+assert.sameValue(log.length, 0, "log.length");
+
+asyncIterator.next("next-arg-1").then(v => {
+  assert.sameValue(log[0].name, "before yield*");
+
+  assert.sameValue(log[1].name, "get [Symbol.asyncIterator]");
+
+  assert.sameValue(log[2].name, "call [Symbol.asyncIterator]");
+  assert.sameValue(log[2].args.length, 0, "[Symbol.asyncIterator] args.length");
+
+  assert.sameValue(log[3].name, "get next");
+
+  assert.sameValue(log[4].name, "call next");
+  assert.sameValue(log[4].args.length, 1, "next args.length");
+  assert.sameValue(log[4].args[0], undefined, "next args[0]");
+
+  assert.sameValue(log[5].name, "get next then (1)");
+
+  assert.sameValue(log[6].name, "call next then (1)");
+  assert.sameValue(log[6].args.length, 2, "next then args.length");
+  assert.sameValue(typeof log[6].args[0], "function", "next then args[0]");
+  assert.sameValue(typeof log[6].args[1], "function", "next then args[1]");
+
+  assert.sameValue(log[7].name, "get next done (1)");
+
+  assert.sameValue(log[8].name, "get next value (1)");
+
+  assert.sameValue(log[9].name, "get next done (1)");
+
+  assert.sameValue(v.value, "next-value-1");
+  assert.sameValue(v.done, false);
+
+  assert.sameValue(log.length, 10, "log.length");
+
+  asyncIterator.next("next-arg-2").then(v => {
+    assert.sameValue(log[10].name, "get next");
+
+    assert.sameValue(log[11].name, "call next");
+    assert.sameValue(log[11].args.length, 1, "next args.length");
+    assert.sameValue(log[11].args[0], "next-arg-2", "next args[0]");
+
+    assert.sameValue(log[12].name, "get next then (2)");
+
+    assert.sameValue(log[13].name, "call next then (2)");
+    assert.sameValue(log[13].args.length, 2, "next then args.length");
+    assert.sameValue(typeof log[13].args[0], "function", "next then args[0]");
+    assert.sameValue(typeof log[13].args[1], "function", "next then args[1]");
+
+    assert.sameValue(log[14].name, "get next done (2)");
+
+    assert.sameValue(log[15].name, "get next value (2)");
+
+    assert.sameValue(log[16].name, "after yield*");
+    assert.sameValue(log[16].value, "next-value-2");
+
+    assert.sameValue(v.value, "return-value");
+    assert.sameValue(v.done, true);
+
+    assert.sameValue(log.length, 17, "log.length");
+  }).then($DONE, $DONE);
+}).catch($DONE);
```

Comment:
Meaning, right now, if the promise rejects, the test will just be marked as done. Maybe the test should fail. E.g. maybe `.catch(e => assert(false, "Must not reject"))` is more appropriate?